common.mk

```diff
@@ -0,0 +1,33 @@
+ARCH ?= x86
+OUT_DIR := .out
+OBJS := $(patsubst %.c,$(OUT_DIR)/%.o,$(wildcard *.c))
+DEP := $(OBJS:%.o=%.d)
+BPF_CFLAGS := -target bpf \
+              -Wall \
+              -Wno-unused-value \
+              -Wno-pointer-sign \
+              -Wno-compare-distinct-pointer-types \
+              -fno-stack-protector \
+              -Werror \
+              -D __TARGET_ARCH_$(ARCH) \
+	      -O2
+
+ALL_REQ := $(OBJS)
+
+all: $(ALL_REQ)
+
+$(OUT_DIR):
+	mkdir -p $(OUT_DIR)/
+
+$(OBJS): | $(OUT_DIR)
+
+$(OUT_DIR)/%.o: %.c
+	$(CLANG) $(CFLAGS) $(BPF_CFLAGS) $(INCLUDES_EXTRA) -MMD -c -g -o $@ $<
+	$(OBJCOPY) --strip-debug $@
```

Comment:
Maybe only for `release`? Or is that never useful?

Reply:
this is always useful. It strips dwarf info. BTF info is still there (func info and line info as well). 
You can tell the difference in the binary size (and skel creation) if you remove the line :)

with:
`110K Feb  3 12:35 ./core/probe/kernel/bpf/.out/kretprobe.bpf.o`

without:
`1.3M Feb  3 12:37 ./core/probe/kernel/bpf/.out/kretprobe.bpf.o`

Reply:
Nice space saving. I was wondering if the unstripped version would be useful for debugging. My understanding of the above is it's not, so that's fine, thanks!

Reply:
> Nice space saving. I was wondering if the unstripped version would be useful for debugging. My understanding of the above is it's not, so that's fine, thanks!

yup